Submodule update

pbehne opened the pull request:
Updates application submodules

moosebuild commented:
Job [VTB Documentation](https://civet.inl.gov/job/3168274/), step Sync Docs on 85de702 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/virtualtestbed/docs/PRs/654/site)

This comment will be updated on new commits.

pbehne commented:
Here are the available bison modules on hpc:
```
[behnpa@rod ~]$ module spider bison

-----------------------------------------------------------------------------------------------------------------------
  bison-coupletest-openmpi:
-----------------------------------------------------------------------------------------------------------------------
     Versions:
        bison-coupletest-openmpi/2024.09.27-a169ad1
        bison-coupletest-openmpi/2024.10.12-31e8faf

-----------------------------------------------------------------------------------------------------------------------
  For detailed information about a specific "bison-coupletest-openmpi" package (including how to load the modules) use the module's full name.
  Note that names that have a trailing (E) are extensions provided by other modules.
  For example:

     $ module spider bison-coupletest-openmpi/2024.10.12-31e8faf
-----------------------------------------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------------------------------------
  bison-openmpi:
-----------------------------------------------------------------------------------------------------------------------
     Versions:
        bison-openmpi/2024.08.22-37928d0
        bison-openmpi/2025.03.13-618b661
        bison-openmpi/2025.03.31-cf80d8a
        bison-openmpi/2025.04.12-5ae1f05
        bison-openmpi/2025.04.29-41f37ba
        bison-openmpi/2025.05.12-ce069ab
        bison-openmpi/2025.05.30-f47516a
        bison-openmpi/2025.06.14-4a0d04d
        bison-openmpi/2025.06.19-f718b8e
        bison-openmpi/2025.06.30-e7afde6
        bison-openmpi/2025.07.14-a87760a
        bison-openmpi/2025.07.30-8eed390
        bison-openmpi/2025.08.14-8f19da0
        bison-openmpi/2025.08.27-feec5cf
        bison-openmpi/2025.08.30-6a97bce
        bison-openmpi/2025.09.12-b33bb7c

-----------------------------------------------------------------------------------------------------------------------
  For detailed information about a specific "bison-openmpi" package (including how to load the modules) use the module's full name.
  Note that names that have a trailing (E) are extensions provided by other modules.
  For example:

     $ module spider bison-openmpi/2025.09.12-b33bb7c
-----------------------------------------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------------------------------------
  bison-openmpi-cudatorch:
-----------------------------------------------------------------------------------------------------------------------
     Versions:
        bison-openmpi-cudatorch/2025.03.13-618b661
        bison-openmpi-cudatorch/2025.03.31-cf80d8a
        bison-openmpi-cudatorch/2025.04.12-5ae1f05
        bison-openmpi-cudatorch/2025.04.29-41f37ba
        bison-openmpi-cudatorch/2025.05.12-ce069ab
        bison-openmpi-cudatorch/2025.05.30-f47516a
        bison-openmpi-cudatorch/2025.06.14-4a0d04d

-----------------------------------------------------------------------------------------------------------------------
  For detailed information about a specific "bison-openmpi-cudatorch" package (including how to load the modules) use the module's full name.
  Note that names that have a trailing (E) are extensions provided by other modules.
  For example:

     $ module spider bison-openmpi-cudatorch/2025.06.14-4a0d04d
-----------------------------------------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------------------------------------
  bison-squashfs-openmpi: bison-squashfs-openmpi/2025.08.14-8f19da0
-----------------------------------------------------------------------------------------------------------------------
```

Right now, the submodule updater script takes the hash from the last entry. In this case, it gets `8f19da0` from `bison-squashfs-openmpi: bison-squashfs-openmpi/2025.08.14-8f19da0`. Is this the version we want to be using?

GiudGiud commented:
This one
>         bison-openmpi/2025.09.12-b33bb7c

is probably the better one

pbehne commented:
Okay, I updated the submodule update script to look for openmpi packages only. I also fixed the issue where it was checking for `module spider SAM` instead of `module spider sam`. I don't see any modules for relap on hpc. If this goes well, I'll make a separate PR for the script updates.

GiudGiud commented:
looks like we need to skip a hpmr_h2 tests with BC
new diff with relap7
new error i think with DW?
